### DISCUSSION TITLE
Does Supabase take care of creating database indexes?

### DISCUSSION THREAD

**Discussion by timuster** [supabase/supabase#1500] [Questions] [answered]
Does Supabase take care of creating database indexes, or is it something that developers need to take care of? One of the advantages of using Firebase is that the developer does not have to think about behind-the-scenes database optimization... I was wondering what's recommended for the Supabase PostgreSQL databases. I couldn't find anything in the docs.

**Comment by inian** [comment, marked as answer]
Hi @timuster, we don't add the indexes automatically, since the kind of indexes you need will depend on the workload you are running and adding unnecessary indexes can in fact slow down your write performance. 

Am seeing Firebase documentation and it [requires you to create indexes ](https://firebase.google.com/docs/firestore/query-data/indexing) manually right?
